### DISCUSSION TITLE
Providers to extend initial values from other Providers in the tree

### DISCUSSION THREAD

**Discussion by carlesba** [pmndrs/jotai#1601] [Ideas] [answered]
Currently when setting many nested providers, the initial values are not being extended.

For instance

```jsx

const a = atom(10)
const b = atom(100)

//...
<Provider initialValues={[[a, 20]]}>
 <DisplayAtoms /> // will show a: 20; b: 100
  <Provider initialValues={[[b, 200]]}>
   <DisplayAtoms /> // will show a: 10, b: 200
```

It'd be really handy being able to compose states from higher parts of the tree.

**Comment by dai-shi** [comment, marked as answer]
Hi, thanks for opening the discussion.
I've been struggling with explaining this, but in short, this use case is out of the scope of the core library.
In Jotai v2 API #1515, the `initialValues` prop will no longer be available. We do have a migration path for what we can do. But, composing state from the tree is never supported from the beginning. FYI, `useHydrateAtoms` is some tool trying to overcome in some cases, but not useful in this case.
So, what's the recommendation? This is an open field for third-party libraries (btw, we have many third-party libraries in https://github.com/jotai-labs) and one promising one is [jotai-molecules](https://github.com/saasquatch/jotai-molecules). Please also see #1595.

**Reply by carlesba** [reply]
ありがとうございます！jotai-molecules is definitely what I was looking for. 

I couldn't find the use case for jotai as state is always based on external sources. Molecules could solve this. 

Still figuring out how to move hooks state into atoms. Otherwise everything goes to hooks. 

Thanks again!
